android_vs_ios.md

```diff
@@ -0,0 +1,175 @@
+There are no equivalent of these on iOS.
+
+## Testing
+Android runs [unit tests](https://developer.android.com/training/testing/unit-testing/local-unit-tests) on the computer’s JVM, without Android device or emulator. All Android calls are stubbed out and often just throws an exception when invoked in tests. This means testable classes (Presenter, ViewModels) should not directly include any Android classes/calls. There is a library called [Robolectric](http://robolectric.org/) which tries to reimplement all the Android calls in JVM, but it should be used sparingly since it’s slower and relies too much on “fake” Android code.\
+[Espresso](https://developer.android.com/training/testing/espresso/) is the official framework for UI testing. It can interact and verify UI elements as a user would (clicking, checking text, etc). UI tests are run in a separate process, but they can still directly access the code of the app that’s being tested.\
+[Mockito](https://site.mockito.org/) is widely used for test mocks.
+
+iOS runs [unit tests](https://developer.apple.com/documentation/xctest) on device/simulator, so it’s fine to call iOS methods in unit tests.\
+Apple offers [UI test target](https://developer.apple.com/library/archive/documentation/DeveloperTools/Conceptual/testing_with_xcode/chapters/09-ui_testing.html) for running UI tests. Similar to Espresso, it runs in a separate process and can interact and verify UI elements as a user would (clicking, checking text, etc). However, it cannot access the app code and is meant for blackbox testing. Due to this, it might be more preferable to run UI tests on a unit test target instead of UI test target. The most popular UI testing frameworks ([KIF](https://github.com/kif-framework/KIF) and [EarlGrey](https://github.com/google/EarlGrey/)) are run on unit test target.\
+iOS developers generally write all mocks manually, partly due to lack of a good mocking framework. There is a library called [Cuckoo](https://github.com/Brightify/Cuckoo) which seems to be slightly more limited version of Mockito.
```

Comment:
I'd rather say that there are no *widely-used* mocking frameworks for Swift (no good ones? That's a little harsh :-) ) 

But I think that's mainly because people say that Swift's features (especially extending protocols) make it so *easy* to make a mock by hand that it's *unnecessary* to use a framework.

Maybe someone with a bit more experience on this front (Oh @vinnyt, @vinnyt , wherefore art though @vinnyt ?) could elaborate a bit on this point?

Reply:
I guess no good ones might be a bit too harsh. A better wording would be there isn't one that can match Mockito in its capabilities. I haven't seen any special swift features utilized that helps mocking, other than just using protocols which Java/Kotlin already does with interfaces. I do think that the difference here is more ideological than practical. Using a mocking framework is never necessary (if you are already doing dependency injection correctly), it just makes mocking easier in certain scenarios. The disagreement is on what scenarios it should be applied to, and even what is considered "easier".

Reply:
I'm still pushing for elimination of ", partially due to lack of a good mocking framework."

Reply:
I will replace it with "partially due to lack of a mature and widely-used mocking framework."

Reply:
I will replace it with "partially due to lack of a mature and widely-used mocking framework for Swift."